packages/quantic/force-app/examples/main/lwc/exampleQuanticSort/exampleQuanticSort.js

```diff
@@ -18,12 +57,80 @@ export default class ExampleQuanticSort extends LightningElement {
     },
   ];
 
+  customSortOptions = [
+    {
+      label: 'Date ascending',
+      value: 'date ascending',
+      criterion: {
+        by: 'date',
+        order: 'ascending',
+      },
+    },
+    {
+      label: 'Views Descending',
+      value: '@ytviewcount descending',
+      criterion: {
+        by: 'field',
+        field: 'ytviewcount',
+        order: 'descending',
+      },
+    },
+    {
+      label: 'No Sort',
+      value: 'nosort',
+      criterion: {
+        by: 'nosort',
+      },
+    },
+  ];
+
+  invalidCustomSortOptions = [
+    {
+      label: undefined,
+      value: 'date ascending',
+      criterion: {
+        by: 'nosort',
+      },
+    },
+    {
+      label: 'Date ascending',
+      value: undefined,
+      criterion: {
+        by: 'date',
+        order: 'ascending',
+      },
+    },
+    {
+      label: 'Views Descending',
+      value: '@ytviewcount descending',
+      criterion: {
+        by: 'field',
+        field: undefined,
+        order: 'descending',
+      },
+    },
+  ];
+
   get notConfigured() {
     return !this.isConfigured;
   }
 
+  get customSortOptionsEnabled() {
+    return this.config.customSortOptionsEnabled;
+  }
+
   handleTryItNow(evt) {
     this.config = evt.detail;
     this.isConfigured = true;
   }
+
+  render() {
+    if (this.withInvalidCustomSortOptions) {
+      return templateWithInvalidCustomSortOptions;
+    }
+    if (this.withCustomSortOptions) {
+      return templateWithCustomSortOptions;
+    }
```

Comment:
We don't need two templates to deal with the valid and invalid options. cause they are both doing the exact same thing, the only difference is the object used to displayed the options and this could be easily handled in the JS file.

like the following:
when the add custom options button is clicked do `this.customOptions = validOptions`
and when the add invalid custom options button is clicked do `this.customOptions = invalidOptions`
and then you will have just one template that uses `this.customOptions`